### PR TITLE
Implemented \JsonSerializable

### PR DESCRIPTION
**Pull request by tylercrompton**
This addresses #5 partially or fully depending on what you had in mind. This works great for JSON API's that use this library.

### CODE REVIEW

**Comment by sergeyklay** [comment]
Hi!

@tylercrompton Can you please add `JsonSerializable` interface class for passing php-5.3 tests?

```php
<?php
interface JsonSerializable {
    public function jsonSerialize();
}
```

File: lib/Money/Money.php [hunk 1]
```diff
@@ -297,4 +297,17 @@ public static function stringToUnits( $string )
 
         return (int) $units;
     }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @return array
+     */
+    function jsonSerialize()
```
Comment (sergeyklay) [inline lib/Money/Money.php]: also use visibility here, like `public function jsonSerialize()`  to follow the general coding style and PSR standard

**Comment by tylercrompton** [comment]
Right. I forgot `JsonSerializable` isn't in PHP 5.3. Eh. I would actually consider upgrading the PHP dependency to 5.4+ instead. [PHP 5.3 hasn't even have security support for nearly six months now.](http://php.net/supported-versions.php)

**Comment by sergeyklay** [comment]
To pass the php-5.3 test is sufficient to add the `\JsonSerializable` stub and using `spl_autoload_register` function in phpunit bootstrap file:

```php
spl_autoload_register(function ($class) {
    include 'money/tree/master/tests/' . $class . '.php';
});
```

**Comment by tylercrompton** [comment]
Personally, I prefer not to add support for a version of a language that the implementers no longer support. Besides, the test would pass with aforementioned change but the code would still fail in production. In order to add support, we'd have to provide a custom `json_encode` function that would handle objects implementing `\JsonSerializable`. That just seems like too much to me.

**Comment by mathiasverraes** [comment]
Thanks, see https://github.com/mathiasverraes/money/pull/86

**Comment by mathiasverraes** [comment]
The minimum requirement for the `nextrelease` branch will be php 5.4

**Comment by sergeyklay** [comment]
:+1:

**Pull request closed without merging**